### PR TITLE
Filter result from smart content for pages by audience targeting

### PR DESCRIPTION
**Pull request by danrot**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes parts of #3205 
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR adds a checkbox to decide if audience targeting should be considered. It also adapts the query from the content query builder.

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/smart-content/main.js [hunk 1]
```diff
@@ -1120,7 +1135,19 @@ define(['services/husky/util'], function(util) {
                 }));
             }
 
-            if ($subContainer.find('> *').length > 0 ) {
+            if ($subContainer.find('> *').length >= 2 ) {
```
Comment (chirimoya) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/smart-content/main.js]: remove space in condition

File: src/Sulu/Component/Content/SmartContent/QueryBuilder.php [hunk 2]
```diff
@@ -258,6 +277,33 @@ private function buildDatasourceWhere()
...
     /**
+     * Returns the where part for the audience targeting.
+     *
+     * @param string $locale
+     *
+     * @return string|void
+     */
+    private function buildAudienceTargeting($locale)
+    {
+        $targetGroup = $this->targetGroupEvaluator->evaluate();
```
Comment (chirimoya) [inline src/Sulu/Component/Content/SmartContent/QueryBuilder.php]: is this really the evaluation or those it only return the target group if it is already evaluated?

Comment (danrot) [inline src/Sulu/Component/Content/SmartContent/QueryBuilder.php]: It's the real evaluation, and returns the result of the evaluation, which is a target group.

**Comment by chirimoya** [comment]
ACK

**Comment by danrot** [comment]
@chirimoya Fixed the change detection issue with the last commit, but it is a bit hacky... the problem is that the change detection relies on the value of the data array which will be sent with the URL updating the list of the smart content. I just added the `audienceTargeting` property there, so that the check works correctly. But I also had to remove it before it is sent, because having the `audienceTargeting` value in the URL activates audience targeting for this request, which causes the system to evaluate the current request. This will change the response if a target group is matched...

**Pull request merged**